scripts/pylib/twister/twisterlib/scripting.py

```diff
@@ -0,0 +1,141 @@
+        self.post_script = self._convert_to_script(self.post_script)
+
+    # Converts a dictionary to a Script instance if necessary.
+    def _convert_to_script(self, script: dict | Script | None) -> Script | None:
+        if isinstance(script, dict):
+            return Script(**script)
+        return script
+
+
+@dataclass
+# Holds a collection of scripting elements.
+class ScriptingData:
+    elements: list[ScriptingElement] = field(default_factory=list)
+
+    # Ensures all elements are ScriptingElement instances.
+    def __post_init__(self):
+        self.elements = [
+            elem if isinstance(elem, ScriptingElement) else ScriptingElement(**elem)
+            for elem in self.elements
+        ]
+
+    @classmethod
+    # Loads scripting data from a YAML file.
+    def load_from_yaml(cls, filename: Path | str, schema: dict) -> ScriptingData:
+        try:
+            raw_data = scl.yaml_load_verify(filename, schema) or []
+            return cls(raw_data)
+        except scl.EmptyYamlFileException:
+            logger.error(f'Scripting file {filename} is empty')
+            sys.exit(1)
+        except FileNotFoundError:
+            logger.error(f'Scripting file {filename} not found')
+            sys.exit(1)
+        except Exception as e:
+            logger.error(f'Error loading {filename}: {e}')
+            sys.exit(1)
+
+    # Extends the current scripting data with another set of scripting data.
+    def extend(self, other: ScriptingData) -> None:
+        self.elements.extend(other.elements)
+
+    # Finds a scripting element that matches the given scenario and platform.
+    def find_matching_scripting(self, scenario: str, platform: str) -> ScriptingElement | None:
+        matched_elements = []
+
+        for element in self.elements:
+            if element.scenarios and not _matches_element(scenario, element.scenarios):
+                continue
+            if element.platforms and not _matches_element(platform, element.platforms):
+                continue
+            matched_elements.append(element)
+
+        # Check for override_script
+        override_scripts = [
+            elem
+            for elem in matched_elements
+            if (
+                (elem.pre_script and elem.pre_script.override_script)
+                or (elem.post_flash_script and elem.post_flash_script.override_script)
+                or (elem.post_script and elem.post_script.override_script)
+            )
+        ]
+
+        if len(override_scripts) > 1:
+            logger.error("Multiple override definition for scripts found")
+            sys.exit(1)
+        elif len(override_scripts) == 1:
+            return override_scripts[0]
+        elif matched_elements:
+            return matched_elements[0]
+
+        return None
+
+
+# Checks if the given element matches any of the provided patterns.
+def _matches_element(element: str, patterns: list[str]) -> bool:
+    return any(pattern in element for pattern in patterns)
```

Comment:
```suggestion
    return any(pattern == element for pattern in patterns)
```
otherwise when an element has e.g. just 'mec' in platforms it will match like `*mec*`, isn't it ?